test/release/examples/primers/ranges.chpl

```diff
@@ -7,245 +7,384 @@
+// See the `Range Operators` section below for more information about
+// strided ranges and the ``by`` operator.
+
+/*
+   Uses of Ranges
+   --------------
+*/
 
 //
-// Ranges are a basic building block for iteration.  Here we use a
-// for-loop to iterate over the sequence represented by ``r`` and
-// compute the sum of the indices in the sequence.
+// Ranges are a basic building block for iteration.  The following
+// for-loop iterates over the sequence represented by ``r``, computing
+// the sum of the values in the sequence.
 //
 var sum: int = 0;
-for i in r do
+for i in 1..10 do       // compute 1 + 2 + 3 + ... + 10
   sum += i;
-writeln("The sum of the values in ", r, " is ", sum);
-
-//
-// We can compute this same sum with a reduction over the range.
-//
-sum = + reduce r;
-writeln("The sum of the values in ", r, " as computed by reduce is also ", sum);
+writeln("The sum of the values in '1..10' is ", sum);
 writeln();
 
+// Ranges can also drive parallel loop forms such as
+// :ref:`forall<primers-forallLoops>` and
+// :ref:`coforall<primers-taskparallel-coforall>` loops.
+
 //
-// Ranges are also the building blocks for domains.  Here
-// we use the range ``r`` to build up a 2-D domain and an array.
+// Ranges also serve as the building block for rectangular
+// :ref:`domains <primers-domains>` and :ref:`arrays
+// <primers-arrays>`.  Here we use the range ``r`` to declare a 2D
+// 10x10 domain and an array over it.
 //
 writeln("Domains and arrays");
-const D: domain(2) = {r, r};
-var A: [D] real;
+const D = {1..10, 1..10};  // could also use '= {r, r};'
+var A: [D] real;           // a 10x10 array of real floating point values
 writeln("D = ", D);
 writeln("Array A");
 writeln(A);
 writeln();
 
+// Arrays can also be declared in terms of anonymous domains by using
+// ranges directly (optionally dropping the curly brackets):
+
+var A1: [1..n] int,
+    A2: [1..n, 1..n] string;
+
 //
-// Ranges don't have to be fully bounded.  A range without a low bound
-// represents an infinite sequence of numbers that are less than or equal to
-// the high bound.  Similarly a range without a high bound represents
-// an infinitude of numbers greater or equal to the low bound.  A range
-// with no bounds is allowed as well.
+// Ranges can be used in general contexts that expect an iterable
+// collection of values, such as reductions:
 //
-writeln("Unbounded ranges");
-writeRange(1..);
-writeRange(..5);
-writeRange(..);
+sum = + reduce r;  // compute 1 + 2 + 3 + ... + 10
+writeln("The sum of the values in ", r, ", computed by reduce, is also ", sum);
+
+// Or whole-array operations:
+A1 = r;      // assign the elements of A1 their corresponding values of r
+writeln(A1);
+writeln();
 
+
+/*
+  Variations on Basic Ranges
+  --------------------------
+*/
+
+// Ranges can be defined over boolean values and enumerated types.  In
+// the case of enumerations, the range's values are ordered according
```

Comment:
How about:

```
// Ranges can be defined over boolean values and enumerated types.  In          
// the case of enumerations, the range's sequence of values is based            
// on the declaration order of the enum's constants rather than their           
// (optional) numerical values.                                                 
```

Reply:
Yep, that's better.